express-api/src/services/properties/propertiesServices.ts

```diff
@@ -134,12 +149,35 @@ const collectFindOptions = (filter: PropertyUnionFilter) => {
 };
 
 const getPropertiesUnion = async (filter: PropertyUnionFilter) => {
-  return AppDataSource.getRepository(PropertyUnion).find({
-    where: collectFindOptions(filter),
-    take: filter.quantity,
-    skip: (filter.page ?? 0) * (filter.quantity ?? 0),
-    order: sortKeyMapping(filter.sortKey, filter.sortOrder as FindOptionsOrderValue),
-  });
+  const options = collectFindOptions(filter);
+  const query = AppDataSource.getRepository(PropertyUnion)
+    .createQueryBuilder()
+    .where(
+      new Brackets((qb) => {
+        options.forEach((option) => qb.orWhere(option));
+      }),
+    );
+
+  // Restricts based on user's agencies
+  if (filter.agencyId?.length) {
+    query.andWhere('agency_id IN(:list)', {
+      list: filter.agencyId.join(','),
```

Comment:
I think this actually needs to be `query.andWhere('agency_id IN (:...list)', { list: filter.agencyId } )`.
https://orkhan.gitbook.io/typeorm/docs/select-query-builder#using-parameters-to-escape-data